launcher/modManager/cmodlist.cpp

```diff
@@ -213,6 +213,19 @@ CModEntry CModList::getMod(QString modname) const
 		else
 			settings.insert("active", conf);
 	}
+	
+	if(settings["active"].toBool())
+	{
+		QString rootPath = path.section('/', 0, 1);
+		if(path != rootPath)
+		{
+			conf = getValue(modSettings, rootPath);
+			if(!conf.isNull() && !conf.toMap()["active"].toBool())
+			{
+				settings = conf.toMap();
```

Comment:
please extract `conf.toMap()` into a variable

Reply:
why is purpose?

Reply:
you're doing it twice, might not be cheap if map is big

Reply:
oh, it's returns value, not reference, ok then